src/main/java/ml/shifu/shifu/core/dtrain/gs/GridSearch.java

```diff
@@ -68,10 +80,63 @@
      */
     private int flattenParamsCount;
 
-    public GridSearch(Map<String, Object> rawParams) {
-        assert rawParams != null;
+    public GridSearch(Map<String, Object> rawParams, List<String> configFileContent) {
         this.rawParams = rawParams;
-        parseParams(this.rawParams);
+        List<Map<String, Object>> gridParams = parseParams(configFileContent);
+        if(gridParams != null && gridParams.size() > 0) {
+            assert this.hasHyperParam(rawParams) == false;
+            Set<String> hyperParams = new HashSet<String>();
+            for(Map<String, Object> map: gridParams) {
+                for(Map.Entry<String, Object> gridEntry: map.entrySet()) {
+                    hyperParams.add(gridEntry.getKey());
+                }
+                for(Map.Entry<String, Object> kvEntry: this.rawParams.entrySet()) {
+                    map.put(kvEntry.getKey(), kvEntry.getValue());
+                }
```

Comment:
here parameter set in grid search file will be overrided by in parameters, add check to if not existing in rawParams put t to the map?

Reply:
updated!